File: original_scripts/scala/binary_search/qwen3_1.7b.scala
def binarySearch(arr: Array[Int], target: Int): Int = {
  var low = 0
  var high = arr.length - 1
  while (low <= high) {
    val mid = low + (high - low) / 2
    if (arr(mid) == target) return mid
    if (arr(mid) < target) low = mid + 1
    else high = mid - 1
  }
  -1
}

File: original_scripts/r/binary_search/qwen3_1.7b.r
```cpp
#include <vector>
#include <algorithm>

template <typename T>
void binary_search(std::vector<T>& arr, T target) {
    size_t left = 0;
    size_t right = arr.size();
    
    while (left < right) {
        size_t mid = left + (right - left) / 2;
        if (arr[mid] < target) {
            left = mid + 1;
        } else {
            right = mid;
        }
    }
    
    if (left < arr.size() && arr[left] == target) {
        return;
    }
}
```